### PR TITLE
Switch from Makefile to Earthly

### PR DESCRIPTION
**Pull request by negz**
### Description of your changes

We're trying out a new build tool for one release. See https://github.com/crossplane/crossplane/pull/5711 for context.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `earthly +reviewable` to ensure this PR is ready for review.

### How has this code been tested

[contribution process]: https://git.io/fj2m9

Just `earthly +reviewable` for now.

### CODE REVIEW

File: .github/workflows/promote.yml [hunk 1]
```diff
@@ -1,48 +0,0 @@
-name: Promote
```
Comment (jbw976) [inline .github/workflows/promote.yml:1]: this promote workflow was never used for crossplane-runtime, and therefore safe to delete?

File: .github/renovate.json5 [hunk 2]
```diff
@@ -1,96 +1,208 @@
...
-  "postUpdateOptions": ["gomodTidy"],
-// By default renovate will auto detect whether semantic commits have been used
-// in the recent history and comply with that, we explicitly disable it
-  "semanticCommits": "disabled",
-// All PRs should have a label
-  "labels": ["automated"],
-  "regexManagers": [
+  // The branches renovate should target
+  // PLEASE UPDATE THIS WHEN RELEASING.
+  "baseBranches": [
```
Comment (jbw976) [inline .github/renovate.json5:15]: oh interesting, we weren't specifying the set of `baseBranches` before now - consider making a small tweak to the instructions for updating `baseBranches` in https://github.com/crossplane/release/blob/main/.github/ISSUE_TEMPLATE/release.md, so we remember to update it here too in addition to core crossplane.

File: .github/workflows/renovate.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
+name: Renovate
```
Comment (jbw976) [inline .github/workflows/renovate.yml:1]: wait, was Renovate not running on crossplane-runtime since switching to self hosted?

**Review by jbw976 — APPROVED** [review]
Took this for a quick test drive and all the commands I was running worked successfully! (`+build`, `+reviewable`, `+test`, `+generate`).  Looks like we've incorporated a lot of the lessons from iterating on core crossplane too :)

Just a few questions, nothing blocking on my end!

File: .github/workflows/promote.yml (hunk 1 above)
Comment (negz) [reply .github/workflows/promote.yml:1]: Yeah, exactly. I can't think of what it would have done.

File: .github/workflows/renovate.yml (hunk 3 above)
Comment (negz) [reply .github/workflows/renovate.yml:1]: I think it was still using the Renovate app. See for example https://github.com/crossplane/crossplane-runtime/pull/695 which was opened last week.

I'm guessing we should switch to self hosted for the same reasons as c/c though.

**Pull request merged**